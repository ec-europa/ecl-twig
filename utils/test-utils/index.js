const Twig = require('twig');
const { format } = require('prettier');

const renderTwigFile = (file, options, cb) =>
  Twig.renderFile(file, options, (err, html) =>
    cb(err, format(html, { parser: 'html' }))
  );

const renderTwigFileAsNode = (file, options) =>
  new Promise((resolve, reject) =>
    Twig.renderFile(file, options, (err, html) => {
      if (err) return reject(err);

      const div = document.createElement('div');
      div.innerHTML = html.trim();

      return resolve(div.firstChild);
    })
  );

module.exports = {
  renderTwigFile,
  renderTwigFileAsNode,
};
