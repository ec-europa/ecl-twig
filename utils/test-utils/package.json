{
  "name": "@ecl-twig/test-utils",
  "version": "2.29.0",
  "description": "A local test helper",
  "main": "index.js",
  "private": true,
  "devDependencies": {
    "deepmerge": "4.2.2"
  }
}
