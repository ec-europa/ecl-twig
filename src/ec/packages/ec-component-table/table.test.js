import { merge, renderTwigFileAsNode } from '@ecl-twig/test-utils';

// Import data for tests
import dataDefault from '@ecl/ec-specs-table/demo/data--default';
import dataMulti from '@ecl/ec-specs-table/demo/data--multi';

describe('EC - Table', () => {
  const template = '@ecl-twig/ec-component-table/ecl-table.html.twig';
  const render = params => renderTwigFileAsNode(template, params);

  describe('Default', () => {
    test('renders correctly', () => {
      expect.assertions(1);

      return expect(render(dataDefault)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra class names', () => {
      expect.assertions(1);

      const withExtraClasses = merge(dataDefault, {
        extra_classes: 'custom-table--test',
      });

      return expect(render(withExtraClasses)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra attributes', () => {
      expect.assertions(1);

      const withExtraAttributes = merge(dataDefault, {
        extra_attributes: [
          { name: 'data-test', value: 'data-test-value' },
          { name: 'data-test-1', value: 'data-test-value-1' },
        ],
      });

      return expect(render(withExtraAttributes)).resolves.toMatchSnapshot();
    });
  });

  describe('Zebra', () => {
    test('renders correctly', () => {
      expect.assertions(1);
      dataDefault.zebra = true;
      return expect(render(dataDefault)).resolves.toMatchSnapshot();
    });
  });

  describe('Multi', () => {
    test('renders correctly', () => {
      expect.assertions(1);

      return expect(render(dataMulti)).resolves.toMatchSnapshot();
    });
  });
});
