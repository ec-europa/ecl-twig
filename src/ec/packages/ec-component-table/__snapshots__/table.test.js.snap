// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EC - Table Default renders correctly 1`] = `
<table
  class="ecl-table"
>
  <thead
    class="ecl-table__head"
  >
    <tr
      class="ecl-table__row"
    >
      <th
        class="ecl-table__header"
      >
        Name
      </th>
      <th
        class="ecl-table__header"
      >
        Registration date
      </th>
      <th
        class="ecl-table__header"
      >
        Email
      </th>
    </tr>
  </thead>
  <tbody
    class="ecl-table__body"
  >
    <tr
      class="ecl-table__row"
    >
      <td
        class="ecl-table__cell"
        data-ecl-table-header="Name"
      >
        John
      </td>
      <td
        class="ecl-table__cell"
        data-ecl-table-header="Registration date"
      >
        September 14, 2013
      </td>
      <td
        class="ecl-table__cell"
        data-ecl-table-header="Email"
      >
        <a
          class="ecl-link"
          href="/example"
        >
          john@example.com
        </a>
      </td>
    </tr>
    <tr
      class="ecl-table__row"
    >
      <td
        class="ecl-table__cell"
        data-ecl-table-header="Name"
      >
        Ron
      </td>
      <td
        class="ecl-table__cell"
        data-ecl-table-header="Registration date"
      >
        October 23, 2014
      </td>
      <td
        class="ecl-table__cell"
        data-ecl-table-header="Email"
      >
        <a
          class="ecl-link"
          href="/example"
        >
          ron@example.com
        </a>
      </td>
    </tr>
    <tr
      class="ecl-table__row"
    >
      <td
        class="ecl-table__cell"
        data-ecl-table-header="Name"
      >
        Albert
      </td>
      <td
        class="ecl-table__cell"
        data-ecl-table-header="Registration date"
      >
        December 13, 2014
      </td>
      <td
        class="ecl-table__cell"
        data-ecl-table-header="Email"
      >
        <a
          class="ecl-link"
          href="/example"
        >
          albert@example.com
        </a>
      </td>
    </tr>
    <tr
      class="ecl-table__row"
    >
      <td
        class="ecl-table__cell"
        data-ecl-table-header="Name"
      >
        Marcel
      </td>
      <td
        class="ecl-table__cell"
        data-ecl-table-header="Registration date"
      >
        January 12, 1995
      </td>
      <td
        class="ecl-table__cell"
        data-ecl-table-header="Email"
      >
        <a
          class="ecl-link"
          href="/example"
        >
          marcel@example.com
        </a>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`EC - Table Default renders correctly with extra attributes 1`] = `
<table
  class="ecl-table"
  data-test="data-test-value"
  data-test-1="data-test-value-1"
>
  <thead
    class="ecl-table__head"
  >
    <tr
      class="ecl-table__row"
    >
      <th
        class="ecl-table__header"
      >
        Name
      </th>
      <th
        class="ecl-table__header"
      >
        Registration date
      </th>
      <th
        class="ecl-table__header"
      >
        Email
      </th>
    </tr>
  </thead>
  <tbody
    class="ecl-table__body"
  >
    <tr
      class="ecl-table__row"
    >
      <td
        class="ecl-table__cell"
        data-ecl-table-header="Name"
      >
        John
      </td>
      <td
        class="ecl-table__cell"
        data-ecl-table-header="Registration date"
      >
        September 14, 2013
      </td>
      <td
        class="ecl-table__cell"
        data-ecl-table-header="Email"
      >
        <a
          class="ecl-link"
          href="/example"
        >
          john@example.com
        </a>
      </td>
    </tr>
    <tr
      class="ecl-table__row"
    >
      <td
        class="ecl-table__cell"
        data-ecl-table-header="Name"
      >
        Ron
      </td>
      <td
        class="ecl-table__cell"
        data-ecl-table-header="Registration date"
      >
        October 23, 2014
      </td>
      <td
        class="ecl-table__cell"
        data-ecl-table-header="Email"
      >
        <a
          class="ecl-link"
          href="/example"
        >
          ron@example.com
        </a>
      </td>
    </tr>
    <tr
      class="ecl-table__row"
    >
      <td
        class="ecl-table__cell"
        data-ecl-table-header="Name"
      >
        Albert
      </td>
      <td
        class="ecl-table__cell"
        data-ecl-table-header="Registration date"
      >
        December 13, 2014
      </td>
      <td
        class="ecl-table__cell"
        data-ecl-table-header="Email"
      >
        <a
          class="ecl-link"
          href="/example"
        >
          albert@example.com
        </a>
      </td>
    </tr>
    <tr
      class="ecl-table__row"
    >
      <td
        class="ecl-table__cell"
        data-ecl-table-header="Name"
      >
        Marcel
      </td>
      <td
        class="ecl-table__cell"
        data-ecl-table-header="Registration date"
      >
        January 12, 1995
      </td>
      <td
        class="ecl-table__cell"
        data-ecl-table-header="Email"
      >
        <a
          class="ecl-link"
          href="/example"
        >
          marcel@example.com
        </a>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`EC - Table Default renders correctly with extra class names 1`] = `
<table
  class="ecl-table custom-table--test"
>
  <thead
    class="ecl-table__head"
  >
    <tr
      class="ecl-table__row"
    >
      <th
        class="ecl-table__header"
      >
        Name
      </th>
      <th
        class="ecl-table__header"
      >
        Registration date
      </th>
      <th
        class="ecl-table__header"
      >
        Email
      </th>
    </tr>
  </thead>
  <tbody
    class="ecl-table__body"
  >
    <tr
      class="ecl-table__row"
    >
      <td
        class="ecl-table__cell"
        data-ecl-table-header="Name"
      >
        John
      </td>
      <td
        class="ecl-table__cell"
        data-ecl-table-header="Registration date"
      >
        September 14, 2013
      </td>
      <td
        class="ecl-table__cell"
        data-ecl-table-header="Email"
      >
        <a
          class="ecl-link"
          href="/example"
        >
          john@example.com
        </a>
      </td>
    </tr>
    <tr
      class="ecl-table__row"
    >
      <td
        class="ecl-table__cell"
        data-ecl-table-header="Name"
      >
        Ron
      </td>
      <td
        class="ecl-table__cell"
        data-ecl-table-header="Registration date"
      >
        October 23, 2014
      </td>
      <td
        class="ecl-table__cell"
        data-ecl-table-header="Email"
      >
        <a
          class="ecl-link"
          href="/example"
        >
          ron@example.com
        </a>
      </td>
    </tr>
    <tr
      class="ecl-table__row"
    >
      <td
        class="ecl-table__cell"
        data-ecl-table-header="Name"
      >
        Albert
      </td>
      <td
        class="ecl-table__cell"
        data-ecl-table-header="Registration date"
      >
        December 13, 2014
      </td>
      <td
        class="ecl-table__cell"
        data-ecl-table-header="Email"
      >
        <a
          class="ecl-link"
          href="/example"
        >
          albert@example.com
        </a>
      </td>
    </tr>
    <tr
      class="ecl-table__row"
    >
      <td
        class="ecl-table__cell"
        data-ecl-table-header="Name"
      >
        Marcel
      </td>
      <td
        class="ecl-table__cell"
        data-ecl-table-header="Registration date"
      >
        January 12, 1995
      </td>
      <td
        class="ecl-table__cell"
        data-ecl-table-header="Email"
      >
        <a
          class="ecl-link"
          href="/example"
        >
          marcel@example.com
        </a>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`EC - Table Multi renders correctly 1`] = `
<table
  class="ecl-table"
>
  <thead
    class="ecl-table__head"
  >
    <tr
      class="ecl-table__row"
    >
      <th
        class="ecl-table__header"
      >
        Name
      </th>
      <th
        class="ecl-table__header"
      >
        Registration date
      </th>
      <th
        class="ecl-table__header"
        colspan="3"
      >
        Languages
      </th>
    </tr>
    <tr
      class="ecl-table__row"
    >
      <th
        class="ecl-table__header"
      />
      <th
        class="ecl-table__header"
      />
      <th
        class="ecl-table__header"
      >
        English
      </th>
      <th
        class="ecl-table__header"
      >
        French
      </th>
      <th
        class="ecl-table__header"
      >
        German
      </th>
    </tr>
  </thead>
  <tbody
    class="ecl-table__body"
  >
    <tr
      class="ecl-table__row"
    >
      <td
        class="ecl-table__cell"
        data-ecl-table-header="Name"
      >
        John
      </td>
      <td
        class="ecl-table__cell"
        data-ecl-table-header="Registration date"
      >
        September 14, 2013
      </td>
      <td
        class="ecl-table__cell ecl-table__cell--group"
        data-ecl-table-header="English"
        data-ecl-table-header-group="Language"
      >
        Yes
      </td>
      <td
        class="ecl-table__cell ecl-table__cell--group ecl-table__cell--group"
        data-ecl-table-header="French"
      >
        No
      </td>
      <td
        class="ecl-table__cell ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group"
        data-ecl-table-header="German"
      >
        Yes
      </td>
    </tr>
    <tr
      class="ecl-table__row"
    >
      <td
        class="ecl-table__cell ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group"
        data-ecl-table-header="Name"
      >
        Ron
      </td>
      <td
        class="ecl-table__cell ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group"
        data-ecl-table-header="Registration date"
      >
        October 23, 2014
      </td>
      <td
        class="ecl-table__cell ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group"
        data-ecl-table-header="English"
        data-ecl-table-header-group="Language"
      >
        Yes
      </td>
      <td
        class="ecl-table__cell ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group"
        data-ecl-table-header="French"
      >
        Yes
      </td>
      <td
        class="ecl-table__cell ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group"
        data-ecl-table-header="German"
      >
        Yes
      </td>
    </tr>
    <tr
      class="ecl-table__row"
    >
      <td
        class="ecl-table__cell ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group"
        data-ecl-table-header="Name"
      >
        Albert
      </td>
      <td
        class="ecl-table__cell ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group"
        data-ecl-table-header="Registration date"
      >
        December 13, 2014
      </td>
      <td
        class="ecl-table__cell ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group"
        data-ecl-table-header="English"
        data-ecl-table-header-group="Language"
      >
        No
      </td>
      <td
        class="ecl-table__cell ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group"
        data-ecl-table-header="French"
      >
        No
      </td>
      <td
        class="ecl-table__cell ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group"
        data-ecl-table-header="German"
      >
        Yes
      </td>
    </tr>
    <tr
      class="ecl-table__row"
    >
      <td
        class="ecl-table__cell ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group"
        data-ecl-table-header="Name"
      >
        Marcel
      </td>
      <td
        class="ecl-table__cell ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group"
        data-ecl-table-header="Registration date"
      >
        January 12, 1995
      </td>
      <td
        class="ecl-table__cell ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group"
        data-ecl-table-header="English"
        data-ecl-table-header-group="Language"
      >
        Yes
      </td>
      <td
        class="ecl-table__cell ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group"
        data-ecl-table-header="French"
      >
        Yes
      </td>
      <td
        class="ecl-table__cell ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group ecl-table__cell--group"
        data-ecl-table-header="German"
      >
        Yes
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`EC - Table Zebra renders correctly 1`] = `
<table
  class="ecl-table ecl-table--zebra"
>
  <thead
    class="ecl-table__head"
  >
    <tr
      class="ecl-table__row"
    >
      <th
        class="ecl-table__header"
      >
        Name
      </th>
      <th
        class="ecl-table__header"
      >
        Registration date
      </th>
      <th
        class="ecl-table__header"
      >
        Email
      </th>
    </tr>
  </thead>
  <tbody
    class="ecl-table__body"
  >
    <tr
      class="ecl-table__row"
    >
      <td
        class="ecl-table__cell"
        data-ecl-table-header="Name"
      >
        John
      </td>
      <td
        class="ecl-table__cell"
        data-ecl-table-header="Registration date"
      >
        September 14, 2013
      </td>
      <td
        class="ecl-table__cell"
        data-ecl-table-header="Email"
      >
        <a
          class="ecl-link"
          href="/example"
        >
          john@example.com
        </a>
      </td>
    </tr>
    <tr
      class="ecl-table__row"
    >
      <td
        class="ecl-table__cell"
        data-ecl-table-header="Name"
      >
        Ron
      </td>
      <td
        class="ecl-table__cell"
        data-ecl-table-header="Registration date"
      >
        October 23, 2014
      </td>
      <td
        class="ecl-table__cell"
        data-ecl-table-header="Email"
      >
        <a
          class="ecl-link"
          href="/example"
        >
          ron@example.com
        </a>
      </td>
    </tr>
    <tr
      class="ecl-table__row"
    >
      <td
        class="ecl-table__cell"
        data-ecl-table-header="Name"
      >
        Albert
      </td>
      <td
        class="ecl-table__cell"
        data-ecl-table-header="Registration date"
      >
        December 13, 2014
      </td>
      <td
        class="ecl-table__cell"
        data-ecl-table-header="Email"
      >
        <a
          class="ecl-link"
          href="/example"
        >
          albert@example.com
        </a>
      </td>
    </tr>
    <tr
      class="ecl-table__row"
    >
      <td
        class="ecl-table__cell"
        data-ecl-table-header="Name"
      >
        Marcel
      </td>
      <td
        class="ecl-table__cell"
        data-ecl-table-header="Registration date"
      >
        January 12, 1995
      </td>
      <td
        class="ecl-table__cell"
        data-ecl-table-header="Email"
      >
        <a
          class="ecl-link"
          href="/example"
        >
          marcel@example.com
        </a>
      </td>
    </tr>
  </tbody>
</table>
`;
